deployment/paiLibrary/clusterObjectModel/paiObjectModel.py

```diff
@@ -311,13 +312,26 @@ def getPaiWebPortalUri(self):
     def getK8sApiServerUri(self):
 
         ip = self.rawData["kubernetesConfiguration"]["kubernetes"]["load-balance-ip"]
-        ret = "http://{0}:{1}".format(ip, 8080)
+        ret = ""
+        if self.rawData["kubernetesConfiguration"]["kubernetes"]["k8s-type"] == "aks":
+            ret = "https://{0}:{1}".format(ip, 443)
+        else:
+            ret = "http://{0}:{1}".format(ip, 8080)
         return ret
 
 
 
     def getK8sDashboardUri(self):
 
+        if self.rawData["kubernetesConfiguration"]["kubernetes"]["k8s-type"] == "aks":
+            cmd = "kubectl describe pods kubernetes-dashboard --namespace=kube-system | grep IP"
+            output = linux_shell.execute_shell_with_output(cmd, "Failed to exec {0}".format(cmd))
+            # extract ip address
+            dashboardInfo = re.findall(r"\b(?:[0-9]{1,3}\.){3}[0-9]{1,3}\b", output)
+            if dashboardInfo:
+                ret = "http://{0}:9090".format(dashboardInfo[0])
+                return ret            
+
         vip = ""
```

Comment:
Can this way applies to our on-prem cluster also? If so, suggest to replace the whole logic with the new code universally. We should keep our code simple and unified in maximum extend instead of adding more and more branches unless really no good ways to simplify the code.

Reply:
test it could used for on-prem, unify use the same logic